cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -137,28 +131,29 @@ public void destroy() {
   }
 
   private Properties generateKafkaConfig(int brokerId, String zkConnectStr, String hostname, int port,
-                                         int numPartitions, int replicationFactor, String logDir) {
+                                         CConfiguration cConf) {
     Preconditions.checkState(port > 0, "Port number is invalid.");
+    int numPartitions = cConf.getInt(KafkaConstants.ConfigKeys.NUM_PARTITIONS_CONFIG_DEPRECATED);
     Preconditions.checkState(numPartitions > 0, "Num partitions should be greater than zero.");
 
     Properties prop = new Properties();
+
     prop.setProperty("broker.id", Integer.toString(brokerId));
     if (hostname != null) {
       prop.setProperty("host.name", hostname);
     }
     prop.setProperty("port", Integer.toString(port));
-    prop.setProperty("socket.send.buffer.bytes", "1048576");
-    prop.setProperty("socket.receive.buffer.bytes", "1048576");
-    prop.setProperty("socket.request.max.bytes", "104857600");
-    prop.setProperty("log.dir", logDir);
-    prop.setProperty("num.partitions", Integer.toString(numPartitions));
-    prop.setProperty("log.retention.hours", "24");
-    prop.setProperty("log.flush.interval.messages", "10000");
-    prop.setProperty("log.flush.interval.ms", "1000");
-    prop.setProperty("log.segment.bytes", "536870912");
     prop.setProperty("zookeeper.connect", zkConnectStr);
-    prop.setProperty("zookeeper.connection.timeout.ms", "1000000");
-    prop.setProperty("default.replication.factor", Integer.toString(replicationFactor));
+
+    //ultimately want regex to be "^(kafka\\..server\\.*)$" but for now this gets all deprecated and current config keys
+    //since all deprecated key accesses are redirected to the current config keys in deprecatedKeyMap in Configuration
+    Map<String, String> propConfigs = cConf.getValByRegex("^(kafka\\..*)$");
+    for (Map.Entry<String, String> pair : propConfigs.entrySet()) {
+      String key = pair.getKey();
+      String trimmedKey = key.substring(6);
```

Comment:
Also, if the user uses the deprecated keys (the keys we tell them to use currently, on develop or in most recent release), then it wouldn't start with 'kafka.server.'. So then in that case, we would use substring(6).
This seems problematic.